gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/upgrader/PageHRIDUpgrader.java

```diff
@@ -31,27 +36,47 @@
 @Component
 public class PageHRIDUpgrader implements Upgrader {
 
+    private static final int PAGE_SIZE = 100;
+
     @Lazy
     @Autowired
     private PageRepository pageRepository;
 
     @Override
     public boolean upgrade() {
+        int pageNumber = 0;
+        long updatedCount = 0;
+
         try {
-            pageRepository
-                .findAll()
-                .forEach(page -> {
-                    page.setHrid(page.getId());
-                    try {
+            while (true) {
+                Pageable pageable = new PageableBuilder().pageNumber(pageNumber).pageSize(PAGE_SIZE).build();
+
+                var pagedResult = pageRepository.findAll(pageable);
+                var pages = pagedResult.getContent();
+
+                if (pages.isEmpty()) {
+                    break;
+                }
+
+                for (var page : pages) {
+                    if (!page.getId().equals(page.getHrid())) {
```

Comment:
Since this upgrader is already in production, I would put an if condition like this 
if (page.getHrid() == null) {
  then set the hrid = page.id
}

This way we can make sure that exiting proper HRIDs will not be overridden